userspace/libsinsp/parsers.cpp

```diff
@@ -2307,6 +2334,12 @@ void sinsp_parser::parse_execve_exit(sinsp_evt *evt)
 				*/
 				std::string sdir = parse_dirfd(evt, pathname, dirfd);
 
+				// Update event fdinfo since parse_dirfd is stateless
+				if (sdir != "." && sdir != "<UNKNOWN>")
```

Comment:
Since `parse_dirfd` is now stateless, honor previous behavior by setting the fdinfo when needed (ie: when previous logic set it)